cordova-lib/src/cordova/compile.js

```diff
@@ -40,7 +40,7 @@ module.exports = function compile(options) {
     }).then(function() {
         return hooksRunner.fire('after_compile', options);
     }, function(error) {
-        events.emit('log', 'ERROR building one of the platforms: ' + error + '\nYou may not have the required environment or OS to build this project');
+        events.emit('warn', 'ERROR building one of the platforms: ' + error + '\nYou may not have the required environment or OS to build this project');
```

Comment:
So what's the guidance for error handling? We should emit a warning with detailed error message and then reject the promise with the actual error? I notice below you sometimes create a CordovaError to reject the promise. It will be great if we can come up with consistent set of guidelines for this.